projpicker/web/web.py

```diff
@@ -0,0 +1,176 @@
+    global projpicker_query
+
+    def do_GET(self):
+        if self.path == "/":
+            self.path = "index.html"
+            return http.server.SimpleHTTPRequestHandler.do_GET(self)
+
+        elif self.path == "/ppikdata":
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+            with open("projdata", "r") as f:
+                self.wfile.write(bytes(f.read(), "utf8"))
+
+    def do_POST(self):
+        if self.path == "/data":
+            # http.client.HTTPResponse stores headers and implements
+            # email.message.Message class'
+            # https://docs.python.org/3/library/email.compat32-message.html#email.message.Message
+            header = self.headers
+
+            content_len = int(header.get("content-length"))
+            content_charset = header.get_content_charset(failobj="utf-8")
+            content_bytes = self.rfile.read(content_len)
+            content_body = content_bytes.decode(content_charset)
+            projpicker_query = content_body
+
+            geoms = create_parsable_geoms(json.loads(content_body))
+            crs_list = query(geoms)
+            projpicker_query = bbox_to_json(crs_list)
+
+            projpicker_output_json = open("ppikdata", "w")
+            projpicker_output_json.write(json.dumps(projpicker_query))
+            projpicker_output_json.close()
+
+
+def create_parsable_geoms(geojson):
+    # TODO: Logical operator buttons
+    geoms = geojson["logicalOperator"]
+    for feature in geojson["features"]:
+        json_geom = feature["geometry"]
+        coors = json_geom["coordinates"]
+        geom = Geometry(json_geom["type"], json_geom["coordinates"])
+        geoms += f"\n{geom.type}"
+        if geom.type == "point":
+            geoms += f"\n{geom.coors[0]},{geom.coors[1]}"
+        else:
+            for coors in geom.coors:
+                geoms += f"\n{coors[0]},{coors[1]}"
+
+    return geoms
+
+
+def bbox_to_json(bbox_list):
+    crs_json = {}
+    for crs in bbox_list:
+        entry = {}
+        crs_dict = crs._asdict()
+        for key in list(crs_dict.keys()):
+            entry[key] = crs_dict[key]
+        crs_json[f"{crs.crs_auth_name}:{crs.crs_code}"] = entry
+    return crs_json
+
+
+def query(geoms):
+    crs_list = []
+    crs = []
+    if geoms is not None:
+        if verbose:
+            ppik.message(f"{Color.BOLD}ProjPicker query{Color.ENDC}")
+            ppik.message(f"{Color.BOLD}{'-'*79}{Color.ENDC}")
+            ppik.message(geoms)
+            ppik.message(f"{Color.BOLD}{'-'*79}{Color.ENDC}")
+        parsed_geoms = ppik.parse_mixed_geoms(geoms)
+        crs.extend(ppik.query_mixed_geoms(parsed_geoms))
+        if verbose:
+            ppik.message(f"Query geometries: {parsed_geoms}")
+            ppik.message(f"Number of queried CRSs: {len(crs)}")
+
+    return crs
+
+
+def run(server_class=http.server.HTTPServer,
+        handler_class=HTTPRequestHandler,
```

Comment:
Reasons for these first two arguments when we only support the default classes? Not all HTTP server classes use the syntax `server_class(server_address, handler_class)` (two arguments including a server address in a tuple, and a handler class). If we truly need/want to support third-party user classes, we'll have to take a callback function with those two arguments and call it. Then, the user's callback function will have to handle all this initiation. As is, this code cannot achieve what we want.

Reply:
Not needed anymore because WSGI is now supported.